src/azure-cli-testsdk/azure/cli/testsdk/scenario_tests/base.py

```diff
@@ -163,15 +163,20 @@ def tearDown(self):
     def _save_recording_file(self, *args):  # pylint: disable=unused-argument
         if self.in_recording and self.cassette.dirty:
             self.cassette._save()  # pylint: disable=protected-access
-            if self._outcome.errors or self._outcome.skipped:   # pylint: disable=protected-access
+
+            # Since Python 3.11, unittest.case._Outcome no longer has 'errors' attribute.
+            # The below statements only work in Python <= 3.10.
+            # https://github.com/Azure/azure-cli/issues/27641
+            if hasattr(self._outcome, 'errors') and (self._outcome.errors or self._outcome.skipped):   # pylint: disable=protected-access
                 # remove temp file
                 # to keep the temp_recording_file for debug, a switch logic can add here
                 os.remove(self.temp_recording_file)
-            else:
-                # replace recording_file by temp_recording_file
-                if os.path.exists(self.recording_file):
-                    os.remove(self.recording_file)
-                os.rename(self.temp_recording_file, self.recording_file)
+                return
+
+            # replace recording_file by temp_recording_file
+            if os.path.exists(self.recording_file):
+                os.remove(self.recording_file)
+            os.rename(self.temp_recording_file, self.recording_file)
```

Comment:
To answer @bebound's question, I replace `else` with `return` only to reduce one level of nesting. It is only a code style:

- https://stackoverflow.com/questions/9267643/if-return-vs-if-else-efficiency
- https://stackoverflow.com/questions/9191388/it-is-more-efficient-to-use-if-return-return-or-if-else-return

Reply:
Haha, I'm asking because the `return` gives me a sense of intentional skipping of logic(`extra_c()`).

For example:
``` python
if A:
    condition_a()
    return
condition_b()
extra_c()
```